frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -300,7 +303,9 @@ const commonTemplatesUpdater = ({ id, prevState, dispatch, getState }: UpdateOpt
   const [data, isBase64] = CloudInitDataHelper.getUserData(
     toShallowJS(iGetCommonTemplateCloudInit(iTemplate))?.cloudInitNoCloud,
   );
-  const typeData = CloudInitDataHelper.toCloudInitNoCloudSource(data, isBase64);
+  // generate password
+  const updatedData = data?.replace(/(?<=password:\s+).*/, generateCloudInitPassword());
```

Comment:
you can create an instance of the helper and then rather use has/set/get methods instead

also this will not work with base64 content